src/index.js

```diff
@@ -170,65 +190,51 @@ export const useReduxSelectors = (selectorMap) => {
   const state = store.getState();
   // keys
   const keys = Object.keys(selectorMap);
-  // cache
-  const cacheRef = useRef({
-    selectors: new WeakMap(),
-  });
-  // mapped
+  // lastMapped (ref)
+  const lastMapped = useRef({});
+  // mapped result
   const mapped = createMap(keys, (key) => {
     const selector = selectorMap[key];
-    if (cacheRef.current.selectors.has(selector)) {
-      const partialProxyfied = cacheRef.current.selectors.get(selector);
-      const { proxyfied } = partialProxyfied;
-      const shouldRerunSelector = !proxyEqual(
-        proxyfied.originalState,
-        state,
-        proxyfied.affected,
-      );
-      if (!shouldRerunSelector) {
-        partialProxyfied.resetAffected();
-        return partialProxyfied;
-      }
-    }
-    const proxyfied = createProxyfied(state);
-    const partialState = selector(proxyfied.trappedState);
-    proxyfied.seal(); // do not track any more
-    // if we had `createShallowProxyfied, it should perform much better
-    const partialProxyfied = createProxyfied(partialState);
-    partialProxyfied.proxyfied = proxyfied;
-    return partialProxyfied;
+    const lastResult = lastMapped.current[key];
+    return runSelector(state, selector, lastResult);
   });
+  // if we had `createShallowTrapped, it should perform much better
+  const outerTrapped = createTrapped(createMap(keys, key => mapped[key].value));
   // update ref
-  const lastProxyfied = useRef(null);
+  const lastTracked = useRef(null);
   useLayoutEffect(() => {
+    lastMapped.current = mapped;
     const affected = [];
     keys.forEach((key) => {
-      const selector = selectorMap[key];
-      const partialProxyfied = mapped[key];
-      cacheRef.current.selectors.set(selector, partialProxyfied);
-      if (!partialProxyfied.proxyfied.affected) {
-        partialProxyfied.proxyfied.affected = partialProxyfied.proxyfied.getAffected();
-      }
-      if (partialProxyfied.getAffected().length) {
-        affected.push(...partialProxyfied.proxyfied.affected);
+      if (outerTrapped.affected.indexOf(`.${key}`) >= 0) {
+        const { innerTrapped } = mapped[key];
+        affected.push(...innerTrapped.affected);
```

Comment:
It would be great to create `affectedChunks` as a array of `innerTrapped.affected`, later compare it to `lastAffectedChunks`, and create new `affected` only if chunks are different.
You might be able to skip this operation, and speed up the rest keeping ref equitability.

Reply:
Wow, okay I get that.

Reply:
94189c14328e8590cbe3e846710f3bd9c2221ae6

Reply:
Awesome!